webapp/src/i18n.tsx

```diff
@@ -36,14 +36,20 @@ export function getMessages(lang: string): {[key: string]: string} {
     return messages_en
 }
 
-export function getCurrentLanguage(): string {
-    let lang = localStorage.getItem('language')
+export function getCurrentLanguage(initialLang: string|null): string {
+    let lang = initialLang
+    if (!lang) {
+        lang = localStorage.getItem('language')
+    }
     if (!lang) {
         lang = navigator.language.split(/[-_]/)[0]
```

Comment:
What happens if navigator.language is not one of the supported languages? Does it just default to English? I think it does, but should smoke test this.